app/models/user.rb

```diff
@@ -3,7 +3,8 @@ class User < ActiveRecord::Base
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
-  has_many :metrics
+  
+  has_many :metrics, through: :features
```

Comment:
@eliotsykes Yes! Amazingly. Can you believe after my "polymorphism" meltdown that I actually set up has-many-through? It's actually really easy for this use case, where one has-many is just nested inside another.